File: client/src/components/Canvas.js
import React from 'react'

export const Canvas = ({id, canvasData, width, height, style, onMouseDown, onMouseMove, onMouseUp, onMouseLeave, onMouseEnter}) => {

    if (canvasData.src) {
        
    }

    return (
        <canvas 
            id={id}
            ref={(canvas) => this.canvas = canvas}
            width={width}
            height={height}
            style={style}
            onMouseDown={(e) => onMouseDown(e, id)}
            onMouseMove={(e) => onMouseMove(e, id)}
            onMouseUp={(e) => onMouseUp(e, id)}
            onMouseLeave={(e) => onMouseLeave(e, id)}
            onMouseEnter={(e) => onMouseEnter(e, id)}
            >
        </canvas>
    )

}

File: server/canvas.go
package main

type Canvas struct {
	id int
	src string
}

func getInitialPaintState() []Canvas {

	canvases := make([]Canvas, 0)

	for i := 0; i < 128; i++ {
		canvas := Canvas{id: i, src: ""}
		canvases = append(canvases, canvas)
    }

	return canvases

}